lib/src/utils.dart

```diff
@@ -504,11 +504,27 @@ String prefixLines(String text, {String prefix = '| ', String? firstPrefix}) {
   return lines.join('\n');
 }
 
-/// The subset of strings that don't need quoting in YAML.
-///
-/// This pattern does not strictly follow the plain scalar grammar of YAML,
-/// which means some strings may be unnecessarily quoted, but it's much simpler.
-final _unquotableYamlString = RegExp(r'^[a-zA-Z_-][a-zA-Z_0-9-]*$');
+/// Does this string need quoting in yaml?
+///
+/// This mechanism does not strictly follow the plain scalar grammar of YAML,
+/// which means some strings may be unnecessarily quoted.
+bool _needsYamlQuotes(String s) {
+  // These must be quoted
+  if ([
```

Comment:
List can be `const`. (Small save, but save, over creating a new list on every invocation.)